### ISSUE TITLE
RT-AC88U (HW: A6) regression in 22.03.3

### ISSUE THREAD

**Issue by lightoze** [openwrt/openwrt#11821]
### Describe the bug

At least a few people already experienced boot loop after installing or upgrading 22.03.3. The only workaround seems to be rolling back to 22.03.2, which works fine.

Related forum topic: https://forum.openwrt.org/t/asus-rt-ac88u-hw-a6-broken-in-22-03-3/147882

### OpenWrt version

22.03.3

### OpenWrt target/subtarget

bcm53xx

### Device

Asus RT-AC88U

### Image kind

Official downloaded image

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by wdehoog** [comment]
I've made a log on the serial console. See https://pastebin.com/bMfNcq4t

**Comment by wdehoog** [comment]
Bisecting indicates:
```
b66037f53bf3d943e23cad01986ba079cecb41cd is the first bad commit
commit b66037f53bf3d943e23cad01986ba079cecb41cd
Author: Rafał Miłecki <rafal@milecki.pl>
Date:   Thu Nov 10 09:28:23 2022 +0100

```
This one https://github.com/openwrt/openwrt/commit/b66037f53bf3d943e23cad01986ba079cecb41cd

**Comment by arinc9** [comment]
Quoting myself from OpenWrt forum:

I've pinpointed the offending commit to:

https://github.com/openwrt/openwrt/commit/94372ab6e4f05d90e663c3b648d6048bc49afd60#diff-dbc4ab11d3c2f69fb40f6fbbbffdae7ecab71dcb21d7023d6712c6d29bff5e30

@rmilecki looks like something you changed breaks nvmem. Or parsing nvmem-cells from dt binding weren't there before this commit, and now it's there but won't work properly.

Tested on the 22.03 branch, on the tree at this commit.

**Comment by Djfe** [comment]
@rmilecki what's the state of this? :)

**Comment by michealch** [comment]
22.03.3, 22.03.4, and 22.03.5 are identical and all have bootloop.

**Comment by theyo-tester** [comment]
> 22.03.3, 22.03.4, and 22.03.5 are identical and all have bootloop.

Yes, I can confirm this too....

**Comment by floe** [comment]
Same here, with 22.03.5. Not 100% about the hardware revision, but got it recently at a sale, so probably the latest.

**Comment by floe** [comment]
Yes, confirmed, the label has "HW Ver: A6".

**Comment by arinc9** [comment]
Fixed with https://git.openwrt.org/?p=openwrt/openwrt.git;a=commitdiff;h=bcd76dc7d825b39e2585f1cc165b157a4fb8daa4.

**Comment by floe** [comment]
Excellent, thank you @arinc9 and @rmilecki ! Will this patch be part of 22.03.6 (or 22.04, whatever comes first)?

**Comment by arinc9** [comment]
It'd be a good idea to do that, and for OpenWrt 23.05. Rafal will probably do it if they haven't done it already.

**Comment by rmilecki** [comment]
I wanted to give my patch a few days in the `master` branch first. Today I sent patch for upstream. Give it few days for reviews. I'll update 22.03 and 23.05 after that.

**Comment by fluttirci** [comment]
I have this problem as well, only 22.03.2 works. 

I tried 23.05 versions, 23.05.2 too. They work, but I can't connect to internet, I am using PPPOE vlan id 35. Same settings work on 22.03.2. This is probably same problem I had with ASUS RT-N18U, I had problems with that too on 23.05.2, it also uses bcm53xx driver. See this topic: https://github.com/openwrt/openwrt/issues/13691

**Comment by arinc9** [comment]
> I have this problem as well

No, you don't. This is a different issue which is resolved.

@lightoze please close this resolved issue.

**Comment by rmilecki** [comment]
Fixed in 23.05 with commit 41f27bbb6d0a ("bcm53xx: add the latest fix version of brcm_nvram")

**Comment by rmilecki** [comment]
Now fixed in 22.03 too with commit 9b7311d4f946 ("bcm53xx: add the latest fix version of brcm_nvram")
